### PR TITLE
FPS field for VideoLayer

### PR DESCRIPTION
**Pull request by davidzhao**
Ingress uses this field during stream upload

### CODE REVIEW

File: livekit_models.proto [hunk 1]
```diff
@@ -117,6 +117,8 @@ message VideoLayer {
   // target bitrate, server will measure actual
   uint32 bitrate = 4;
   uint32 ssrc = 5;
+  // publishing FPS for this layer, used with simulcasted streams
+  float fps = 6;
```
Comment (boks1971) [inline livekit_models.proto:121]: thinking about using this with temporal layers?

Comment (boks1971) [reply livekit_models.proto:121]: Why float?

**Review by boks1971 — APPROVED** [review]

File: livekit_models.proto (hunk 1 above)
Comment (davidzhao) [reply livekit_models.proto:121]: The intention is for ingress to reuse this structure when specifying desired layers. I think it'll be hard for clients to determine what fps temporal layers would have.

I had it as a float due to NTSC input streams having 29.97 fps. That's really the only reason :(. Any ideas on how we can better handle this case? GStreamer uses a fraction for fps. wdyt @cnderrauber @frostbyte73 ?

Comment (boks1971) [reply livekit_models.proto:121]: Temporal layers are hard for server too :-) Have to add a frame rate measurement at temporal layer level to do it. I think fraction is fine. I just wanted to understand the reason behind a float. Not sure how exactly we can achieve fractional rate, but having it as a float is not an issue I think.

Comment (frostbyte73) [reply livekit_models.proto:121]: the fraction is just as annoying imo, I had to do this to convert
https://github.com/livekit/server-sdk-go/blob/media/pkg/media/framerate/framerate.go

**Review by frostbyte73 — APPROVED** [review]

**Comment by davidzhao** [comment]
closed in favor of #308

**Comment by boks1971** [comment]
oh wow, totally forgot about this. Maybe, I should change #308 to float too.

**Pull request closed without merging**